VariableAnalysis/Lib/Helpers.php

```diff
@@ -1535,62 +1535,105 @@ public static function getForLoopForIncrementVariable($stackPtr, $forLoops)
 	 */
 	public static function isConstructorPromotion(File $phpcsFile, $stackPtr)
```

Comment:
@sirbrillig Just out of curiousity: why are you not using the `File::getMethodParameters()` to parse the function signature ? That would allow for just checking if the parameter has a `'property_visibility'` index to know whether it is a promoted property...

Reply:
If you didn't know about this one... you might also be interested in the `File::getMemberProperties()` (for declared, non constructor promoted, properties in a class)...

And yes, PHPCSUtils has those too and improved on them further + much more.

Happy to talk things through with you at some point if you think that would help. Unfortunately won't have any time in the foreseeable future to actually help you with the code changes.